ddtrace/internal/telemetry/writer.py

```diff
@@ -121,38 +136,53 @@ def __eq__(self, other):
 
 
 class _TelemetryClient:
-    def __init__(self, endpoint):
-        # type: (str) -> None
-        self._agent_url = get_trace_url()
+    def __init__(self, endpoint, agentless=False):
+        # type: (str, bool) -> None
+        self._telemetry_url = _get_agentless_telemetry_url(config._dd_site) if agentless else get_trace_url()
         self._endpoint = endpoint
         self._encoder = JSONEncoderV2()
+        self._is_agentless = agentless
+        self._is_disabled = False
+
         self._headers = {
             "Content-Type": "application/json",
             "DD-Client-Library-Language": "python",
             "DD-Client-Library-Version": _pep440_to_semver(),
         }
 
+        if self._is_agentless:
+            if not config._dd_api_key:
+                log.debug("Disabling telemetry: no Datadog API key found in agentless mode")
+                self._is_disabled = True
+                return
+            self._headers["dd-api-key"] = config._dd_api_key
+
     @property
     def url(self):
-        return parse.urljoin(self._agent_url, self._endpoint)
+        return parse.urljoin(self._telemetry_url, self._endpoint)
 
     def send_event(self, request: Dict) -> Optional[httplib.HTTPResponse]:
         """Sends a telemetry request to the trace agent"""
+        if self._is_disabled:
+            log.debug("Telemetry is disabled, not sending event.")
+
         resp = None
         conn = None
+        endpoint_str = "Datadog endpoint" if self._is_agentless else "Datadog Agent"
         try:
             rb_json = self._encoder.encode(request)
             headers = self.get_headers(request)
             with StopWatch() as sw:
-                conn = get_connection(self._agent_url)
+                conn = get_connection(self._telemetry_url)
                 conn.request("POST", self._endpoint, rb_json, headers)
                 resp = get_connection_response(conn)
+                # print("sent %d in %.5fs to %s. response: %s", len(rb_json), sw.elapsed(), self.url, resp.status)
```

Comment:
```suggestion
```

Reply:
Delete in [af51bd3](https://github.com/DataDog/dd-trace-py/pull/8948/commits/af51bd374a4aba7bc6393a2390d231fb62a54664).